crates/store/src/state.rs

```diff
@@ -926,14 +1092,14 @@ impl State {
     ) -> Result<AccountProofResponse, DatabaseError> {
         let AccountProofRequest { block_num, account_id, details } = account_request;
 
-        if details.is_some() && !account_id.has_public_state() {
+        if details.is_some() && !account_id.is_public() {
```

Comment:
Why did we change `account_id.has_public_state()` to `account_id.is_public()`? As mentioned in the previous comment, `has_public_state()` is correct.